demisto_sdk/commands/pre_commit/hooks/docker.py

```diff
@@ -0,0 +1,305 @@
+import functools
+import os
+import time
+from collections import defaultdict
+from copy import deepcopy
+from pathlib import Path
+from typing import Dict, Iterable, List, Optional, Set, Tuple
+
+from docker.errors import DockerException
+
+from demisto_sdk.commands.common.constants import TYPE_PWSH, TYPE_PYTHON
+from demisto_sdk.commands.common.content_constant_paths import CONTENT_PATH, PYTHONPATH
+from demisto_sdk.commands.common.docker_helper import (
+    docker_login,
+    get_docker,
+    init_global_docker_client,
+)
+from demisto_sdk.commands.common.native_image import (
+    NativeImageConfig,
+    ScriptIntegrationSupportedNativeImages,
+)
+from demisto_sdk.commands.common.tools import get_yaml, logger
+from demisto_sdk.commands.lint.linter import DockerImageFlagOption
+from demisto_sdk.commands.pre_commit.hooks.hook import Hook
+
+NO_CONFIG_VALUE = None
+
+
+@functools.lru_cache
+def get_docker_python_path() -> str:
+    """
+    precommit by default mounts the content repo to source.
+    This means CommonServerPython's path is /src/Packs/Base/...CSP.py
+    Returns: A PYTHONPATH formatted string
+    """
+    path_to_replace = str(Path(CONTENT_PATH))
+    docker_path = [str(path).replace(path_to_replace, "/src") for path in PYTHONPATH]
+    path = ":".join(sorted(docker_path))
+    logger.debug(f"pythonpath in docker being set to {path}")
+    return path
+
+
+def with_native_tags(
+    tags_to_files: Dict[str, List[Tuple[str, Dict]]], docker_image_flag: str
+) -> Dict[str, List[Tuple[str, Dict]]]:
+    """
+    Adds the native image images into the dict with the files that should be run on them
+    Args:
+        tags_to_files: Dict[str, Tuple[str, dict] the incoming dict without native image of files split according to the docker images
+        docker_image_flag: the flag from the config file. all/native:ga/native:maintenance etc
+
+    Returns: The updated dict with the native images.
+
+    """
+    docker_flags = set(docker_image_flag.split(","))
+    all_tags_to_files = defaultdict(list)
+    native_image_config = NativeImageConfig.get_instance()
+
+    for image, scripts in tags_to_files.items():
+        for file, yml in scripts:
+
+            supported_native_images = ScriptIntegrationSupportedNativeImages(
+                _id=yml.get("commonfields", {}).get("id", ""),
```

Comment:
The precommit command already parses all the integrations as objects.
You need to provide the objects to the docker hook and use `obj.object_id`.